tests/v2/validation/networking/connectivity/network_policy.go

```diff
@@ -0,0 +1,66 @@
+package connectivity
+
+import (
+	"github.com/rancher/shepherd/extensions/workloads"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	pingPodProjectName = "ping-project"
+
+	containerName  = "test1"
+	containerImage = "ranchertest/mytestcontainer"
+)
+
+type resourceNames struct {
+	core   map[string]string
+	random map[string]string
+}
+
+// newNames returns a new resourceNames struct
+// it creates a random names with random suffix for each resource by using core and coreWithSuffix names
+func newNames() *resourceNames {
```

Comment:
This is cool! Likely could have uses outside of just this. Maybe eventually would make sense to port to Shepherd and use it across other tests too. Nice! (I don't want any changes here right now, just wanted to comment that I like this)

Reply:
we can just move this to actions so that it can be used across other tests. [doesn't have to be in this PR].